op-chain-ops/cmd/check-derivation/main.go

```diff
@@ -0,0 +1,432 @@
+	privateKey, err := crypto.HexToECDSA(cliCtx.String("private-key"))
+	if err != nil {
+		return nil, fmt.Errorf("invalid private key: %w", err)
+	}
+	return privateKey, nil
+}
+
+func getSenderAddress(privateKey *ecdsa.PrivateKey) (common.Address, error) {
+	publicKey := privateKey.Public()
+	publicKeyECDSA, ok := publicKey.(*ecdsa.PublicKey)
+	if !ok {
+		return common.Address{}, fmt.Errorf("error casting public key to ECDSA")
+	}
+	from := crypto.PubkeyToAddress(*publicKeyECDSA)
+	return from, nil
+}
+
+// getRandomSignedTransaction returns signed tx which sends 1 wei to random address
+func getRandomSignedTransaction(ctx context.Context, ethClient *ethclient.Client, rng *rand.Rand, from common.Address, privateKey *ecdsa.PrivateKey, chainId *big.Int, txType int, protected bool) (*types.Transaction, error) {
+	randomAddress := getRandomAddress(rng, 0xDEADBEEF)
+	amount := big.NewInt(1)
+	nonce, err := ethClient.PendingNonceAt(ctx, from)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get nonce: %w", err)
+	}
+
+	gasPrice, err := ethClient.SuggestGasPrice(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to suggest gas price: %w", err)
+	}
+	data := testutils.RandomData(rng, 10)
+	var txData types.TxData
+	switch txType {
+	case types.LegacyTxType:
+		gasLimit, err := core.IntrinsicGas(data, nil, false, true, true, false)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get intrinsicGas: %w", err)
+		}
+		txData = &types.LegacyTx{
+			Nonce:    nonce,
+			GasPrice: gasPrice,
+			Gas:      gasLimit,
+			To:       &randomAddress,
+			Value:    amount,
+			Data:     data,
+		}
+	case types.AccessListTxType:
+		accessList := types.AccessList{types.AccessTuple{
+			Address:     randomAddress,
+			StorageKeys: []common.Hash{common.HexToHash("0x1234")},
+		}}
+		gasLimit, err := core.IntrinsicGas(data, accessList, false, true, true, false)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get intrinsicGas: %w", err)
+		}
+		txData = &types.AccessListTx{
+			ChainID:    chainId,
+			Nonce:      nonce,
+			GasPrice:   gasPrice,
+			Gas:        gasLimit,
+			To:         &randomAddress,
+			Value:      amount,
+			AccessList: accessList,
+			Data:       data,
+		}
+	case types.DynamicFeeTxType:
+		gasLimit, err := core.IntrinsicGas(data, nil, false, true, true, false)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get intrinsicGas: %w", err)
+		}
+		gasTipCap, err := ethClient.SuggestGasTipCap(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get gas tip cap: %w", err)
+		}
+		txData = &types.DynamicFeeTx{
+			ChainID:   chainId,
+			Nonce:     nonce,
+			GasTipCap: gasTipCap,
+			GasFeeCap: gasPrice,
+			Gas:       gasLimit,
+			To:        &randomAddress,
+			Value:     amount,
+			Data:      data,
+		}
+	default:
+		return nil, fmt.Errorf("unsupported tx type: %d", txType)
+	}
+
+	tx := types.NewTx(txData)
+
+	signer := types.NewLondonSigner(chainId)
```

Comment:
why not latest signer?

Reply:
The purpose of this script is to test batch submission and derivation for all possible transaction types. 
So for now, using London signer is enough because we don't have a plan to support blob tx on L2.
If we start to support a new TX type on L2 in the future, then we should add a new TX type in this script. To do this, I think it could be better to use an explicit version of Signer.